fe/fe-core/src/main/java/com/starrocks/http/rest/LoadAction.java

```diff
@@ -170,12 +173,22 @@ private void processNormalStreamLoad(
                     nodeIds.add(nodeId);
                 }
             }
-            Collections.shuffle(nodeIds);
         } else {
             SystemInfoService systemInfoService = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo();
-            nodeIds = systemInfoService.getNodeSelector().seqChooseBackendIds(1, true, false, null);
+            nodeIds = systemInfoService.getNodeSelector()
+                                       .seqChooseBackendIds(systemInfoService.getAvailableBackends().size(), true, false, null);
         }
 
+        if (Config.enable_black_list_for_stream_load) {
```

Comment:
`TransactionLoadAction` should also consider black list

Reply:
updated